### PR TITLE
Fix windows builds

### PR DESCRIPTION
**Pull request by Prince213**
This should solve https://github.com/hoelzro/p6-linenoise/issues/29 and have no side-effects on non-Windows OSes.

### CODE REVIEW

File: fill-constants.pl [hunk 1]
```diff
@@ -23,6 +23,9 @@ sub MAIN {
                 $new-line.substr-rw($<value>.from, $<value>.chars) = '0x%x'.sprintf(%constants{$ident});
             }
         }
+        elsif $line.contains('#`(FILL-THIS)') && $*DISTRO.is-win {
```
Comment (JJ) [inline fill-constants.pl]: This line seems to be erroring, and the one below too. Can you please check?

**Review by JJ — CHANGES_REQUESTED** [review]
I don't see anything wrong, but Travis seems to. Can you please check anyway?

File: lib/Linenoise.pm6.in [hunk 2]
```diff
@@ -12,7 +12,19 @@ module Linenoise:ver<0.1.1>:auth<github:hoelzro> {
     my constant O_NONBLOCK   = #`(FILL-ME-IN);
     my constant CLIB         = #`(FILL-ME-IN);
 
-    my constant LIBLINENOISE = %?RESOURCES<libraries/linenoise>.Str;
+    my constant LIBLINENOISE = do if $*DISTRO.is-win {
+        my constant LIBLINENOISE = do {
+            my Str $site-path=#`(FILL-THIS);
```
Comment (JJ) [inline lib/Linenoise.pm6.in]: This line fails to compile, and it actually looks like an error. Maybe missing quotes?

**Review by JJ — CHANGES_REQUESTED** [review]
OK, I found it :-) Can you please check that?

File: lib/Linenoise.pm6.in (hunk 2 above)
Comment (Prince213) [reply lib/Linenoise.pm6.in]: Hmm. It compiles on Windows without errors. What is the error?

**Comment by JJ** [comment]
It's still erroring in Travis, which is a non-windows build: https://travis-ci.org/hoelzro/p6-linenoise/builds/615296812?utm_source=github_status&utm_medium=notification

**Comment by JJ** [comment]
Same problem, I'm afraid... I'm not sure it's your patch's fault, though. It might be related to #29. Did you test it in Windows?

**Comment by Prince213** [comment]
Yes, It installs on Windows without error.

**Comment by JJ** [comment]
I'll have to add an AppVeyor test for Windows... Anyway, the problem is
still that if it breaks in Linux, I can't really accept it...
If you can check what's wrong with it there, please do it. If not, I'll
leave this PR on standby until I find some time to accept and immediately
fix the Linux part...

**Comment by Prince213** [comment]
It should work now. I have tested on my Manjaro machine with a fresh build of rakudo version 2019.07.1 and no error occured.

**Comment by Prince213** [comment]
@JJ Done.

**Comment by ugexe** [comment]
This PR is wrong and should have not been merged. Also just because it “installs on windows” does not mean it works on windows.

**Comment by JJ** [comment]
Any help is really appreciated and welcome. Please raise an issue with the
problems with this PR and I'll try to address them, as well as the author,
if they're available.

**Comment by ugexe** [comment]
If you want help then don’t merge code you do not understand or hasn’t been reviewed by someone who does. Just merging whatever feels good doesn’t actually help and actually makes it harder to find the root cause. I’d get reprimanded for that at a job. I mean you merged code that does

my constant foo = do { my constant foo = ... }

so obviously it was not reviewed at all beyond looking at what Travis says. Speaking of, have you checked to see what the actual tests getting run are to see if that is even a valid benchmark for judging the merit of this code? Because loading the module is not a real test and shouldn’t be considered all that’s needed for a go/no-go. Did anyone actually use this on windows and verified it worked?

**Comment by JJ** [comment]
Did you? If that's the case, can you please report any error it raised?
I hear your concerns, and rest assured we are not going to release anything
before it's tested thoroughly.

**Comment by ugexe** [comment]
I didn’t merge or leave a review — it doesn’t apply to me. When you do nonsense like this (it’s not the first time) it discourages me from doing anything — I have no desire to be associated with that type of sloppy dev behavior. Fucking things up more is NOT a way to garner my help.

**Comment by JJ** [comment]
@ugexe I've patiently taken with as thick a skin as possible all the abuse
you've piled on me and the things I do, trying to get some constructive
things out of them. Sadly, I draw the line at the "nonsense" and at the
"fucking things up" things. I'm going to report your behavior, so that this
kind of things do not happen to me or anyone else in the future. Raku
should be a welcoming community, and getting harassed for doing something
wrong is not something I would wish on anyone else on the community. We
don't have a code of conduct here, so I'll see how to proceed in this case.

**Pull request merged**